libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts

```diff
@@ -150,4 +151,25 @@ export class SegmentedControlComponent<TItem extends SegmentItem = SegmentItem>
       });
     }
   }
+
+  focusNativeButton(event: UIEvent) {
+    (event.target as HTMLIonSegmentButtonElement)?.setFocus();
+  }
+
+  getTabIndex(item: TItem, index: number) {
+    // When focused prevent tab stop from inner native button to outer ion-segment-button:
+    if (this.ionSegmentElement.nativeElement.matches(':focus-within')) return -1;
```

Comment:
Consider extracting to const `segmentElementHasFocusWithin`, `isFocusWithinSegmentElement` or similar